Lib/fontTools/merge.py

```diff
@@ -1048,10 +1155,29 @@ def merge(self, fontfiles):
 
 		return mega
 
-	def _mergeGlyphOrders(self, glyphOrders):
+	def _mergeGlyphOrders(self, glyphOrders, cffTables):
 		"""Modifies passed-in glyphOrders to reflect new glyph names.
 		Returns glyphOrder for the merged font."""
 		mega = {}
+
+		# If we have CFF tables,
+		# rename topDictIndex charStrings
+		for cffTable in cffTables:
+			td = cffTable.cff.topDictIndex[0]
+			d = {}
+			for i, (glyphName, v) in enumerate(td.CharStrings.charStrings.items()):
+				if glyphName in mega:
+					n = mega[glyphName]
+					while (glyphName + "#" + repr(n)) in mega:
+						n += 1
+					mega[glyphName] = n
+					glyphName += "#" + repr(n)
+				mega[glyphName] = 1
+				d[glyphName] = v
+			cffTable.cff.topDictIndex[0].CharStrings.charStrings = d
```

Comment:
I had also to change this code while rebasing to follow the same renaming logic below, but ideally we should de-duplicate this and do the renaming only once.

Reply:
yes, this feels like two methods in one

Reply:
Done.